server/src/main/java/com/cloud/user/DomainManagerImpl.java

```diff
@@ -900,4 +912,171 @@ private void updateDomainChildren(DomainVO domain, String updatedDomainPrefix) {
         }
     }
 
+    @Override
+    @ActionEvent(eventType = EventTypes.EVENT_DOMAIN_MOVE, eventDescription = "moving Domain")
+    @DB
+    public Domain moveDomainAndChildrenToNewParentDomain(MoveDomainCmd cmd) throws ResourceAllocationException {
+        Long idOfDomainToBeMoved = cmd.getDomainId();
+        Long idOfNewParentDomain = cmd.getParentDomainId();
+
```

Comment:
I think we also need to check the access permissions something like

        Account caller = getCaller();
        _accountMgr.checkAccess(caller, domain);

and

       _accountMgr.checkAccess(caller, parentDomain);